pkg/resource/internet_gateway/hooks.go

```
@@ -281,3 +288,192 @@ func updateTagSpecificationsInCreateRequest(r *resource,
+func (rm *resourceManager) associateRouteTable(
+	ctx context.Context,
+	rtID string,
+	igwID string,
+) (err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.associateRouteTable")
+	defer func(err error) {
+		exit(err)
+	}(err)
+
+	input := &svcsdk.AssociateRouteTableInput{
+		RouteTableId: &rtID,
+		GatewayId:    &igwID,
+	}
+
+	_, err = rm.sdkapi.AssociateRouteTableWithContext(ctx, input)
+	rm.metrics.RecordAPICall("UPDATE", "AssociateRouteTable", err)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// disassociateRouteTable will disassociate a RouteTable from a internet
+// gateway based on its association ID, which is given by the API in the
+// output of the Associate* command but can also be found in the
+// description of the route table under the list of its associations.
+func (rm *resourceManager) disassociateRouteTable(
+	ctx context.Context,
+	associationID string,
+) (err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.disassociateRouteTable")
+	defer func(err error) {
+		exit(err)
+	}(err)
+
+	input := &svcsdk.DisassociateRouteTableInput{
+		AssociationId: &associationID,
+	}
+
+	_, err = rm.sdkapi.DisassociateRouteTableWithContext(ctx, input)
+	rm.metrics.RecordAPICall("UPDATE", "DisassociateRouteTable", err)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// getRouteTableAssociations finds all of the route tables that are associated
+// with the current internet gateway and returns a list of each of the
+// association resources.
+func (rm *resourceManager) getRouteTableAssociations(
+	ctx context.Context,
+	res *resource,
+) (assocs []*svcsdk.RouteTableAssociation, err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.getRouteTableAssociations")
+	defer func(err error) {
+		exit(err)
+	}(err)
+
+	input := &svcsdk.DescribeRouteTablesInput{}
+
+	for {
+		resp, err := rm.sdkapi.DescribeRouteTablesWithContext(ctx, input)
+		if err != nil || resp == nil {
+			break
+		}
```

I believe we should just return an error here. The break when `resp.NextToken == nil|""`  is enough

---

I am not fully sure, but if resp is nil ever (without err being nil), then the for loop at line 439 would crash while dereferencing `resp`.